controller: allow RBACSyncConfig to bind to ClusterRole

sbunce opened the pull request:
Issue:
https://github.com/cruise-automation/rbacsync/issues/12

Context:
There are valid use cases for binding a namespaced user or group to a cluster
role. This change removes the validation that was preventing this. See the k8s
documentation on this.
https://kubernetes.io/docs/reference/access-authn-authz/rbac

Testing:
The previous test for binding invalid roles was modified to have a purely made up
role type name. A new test was added for testing binding a cluster role.

mattlandis commented on pkg/controller/controller.go:
```diff
@@ -338,9 +338,11 @@ func (c *Controller) handleConfig(config *rbacsyncv1alpha.RBACSyncConfig) error
 		// need to valdiate that we only create rolebindings with this configuration.
-		if binding.RoleRef.Kind != "Role" {
+		switch binding.RoleRef.Kind {
+		case "Role", "ClusterRole":
```
Can you add a comment here that these are the allowed cases and there is no action to take?  Empty switch statements can feel like bugs without some context.

sbunce replied:
That's a good point. I just removed the comment above the switch and stuck a simple "valid role type" comment in the empty case.

I wonder if I even need this check. I think "Role" and "ClusterRole" are the only two valid role types. I guess it may be paranoid to leave the check in there.

mattlandis replied:
Maybe `// RoleRef types valid in a RoleBinding`?

stevvooe replied:
👍

mattlandis commented:
Can you take a look through the docs and make sure they are accurate after this change.  For instance:

```

For namespace-scoped RBACSyncConfig, the behavior is nearly identical except for the following:

RBACSyncConfig must be defined with a namespace.
RBACSyncConfig can only reference Roles.
All RoleBindings created as a result of the RBACSyncConfig will be in the same namespace.

```

stevvooe commented:
Code/Test LGTM

Please update the readme. Also double check against the [RBAC docs](https://kubernetes.io/docs/reference/access-authn-authz/rbac/) and make sure we are using the same language to describe this scenario.

sbunce commented:
I gave the RBAC docs a re-read.

I updated the README.md to get rid of the point about only being able to reference a role from a RBACSyncConfig. I added an extra paragraph in the README.md about the meaning of referencing a ClusterRole from a RBACSyncConfig. This paragraph is almost identical to what is in the RBAC docs.

Also updated that comment in the controller role ref type check to be more descriptive.

sbunce commented:
I squashed. Ready for merge.
